cmd/soroban-rpc/internal/methods/util.go

```diff
@@ -0,0 +1,42 @@
+package methods
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/stellar/soroban-rpc/cmd/soroban-rpc/internal/db"
+)
+
+func getProtocolVersion(
+	ctx context.Context,
+	ledgerEntryReader db.LedgerEntryReader,
+	ledgerReader db.LedgerReader,
+) (uint32, error) {
+	readTx, err := ledgerEntryReader.NewCachedTx(ctx)
+	if err != nil {
+		return 0, err
+	}
```

Comment:
As I mentioned in a prior review, there is no point int creating a transaction just to obtain the latest ledger sequence. Transactions make sense when querying multiple things from the DB at disjoint times.

Reply:
you can obtain the latest ledger sequence without a transaction, but it would be better if you get the both latest ledger and and the latest ledger sequence within the same transaction

Reply:
Either option is fine but don't start a transaction to query a single thing.